account_due_list/init_hook.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# © 2016 David Dufresne <david.dufresne@savoirfairelinux.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openerp import SUPERUSER_ID
+from openerp.api import Environment
+
+
+logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    """
+    The objective of this hook is to speed up the installation
+    of the module on an existing Odoo instance.
+
+    Without this script, if a database has a few hundred thousand
+    journal entries, which is not unlikely, the update will take
+    at least a few hours.
+
+    The pre init script only writes 0 in the field maturity_residual
+    so that it is not computed by the install.
+
+    The post init script sets the value of maturity_residual.
+    """
+    store_field_stored_invoice_id(cr)
+    store_field_day(cr)
+    store_field_maturity_residual(cr)
+
+
+def post_init_hook(cr, pool):
+    env = Environment(cr, SUPERUSER_ID, {})
+    store_field_maturity_residual_post_init(env)
+
+
+def store_field_stored_invoice_id(cr):
+    cr.execute(
+        """
+        ALTER TABLE account_move_line ADD COLUMN stored_invoice_id integer;
+        COMMENT ON COLUMN account_move_line.stored_invoice_id IS 'Invoice';
+        """)
+
+    logger.info('Computing field stored_invoice_id on account.move.line')
+
+    cr.execute(
+        """
+        UPDATE account_move_line
```

Comment:
@pedrobaeza first, you are missing a join condition. Second, I prefer a subquery. I find it more explicit. I did not try your query. It probably works as well.